src/DynamoCore/Graph/Nodes/NodeModel.cs

```diff
@@ -2344,26 +2344,14 @@ protected virtual ExecutionHints GetExecutionHintsCore()
         ///
         internal void RequestValueUpdate(EngineController engine)
         {
-            // A NodeModel should have its cachedMirrorData reset when it is
-            // requested to update its value. When the QueryMirrorDataAsyncTask
-            // returns, it will update cachedMirrorData with the latest value.
-            //
-            lock (cachedValueMutex)
```

Comment:
can the mutex be removed? What else locks on it? Why remove it?

Reply:
is there behavior that depended on cachedValue being set null?

Reply:
`CachedValue` is still being set to `null` in certain cases, one such case being this one where code is deleted from the CBN, leaving only the function definition in place. This is now happening in line: 2354 below. Earlier only the field was being set to `null` and not the property because of which the property changed event wasn't being raised. As a result, the preview of the node didn't update to `null` upon updating the CBN. 
As `CachedValue` is being set to a non-null value outside of the lock anyway, I thought it would be safe to set it to `null` in cases where the mirror is `null`. Also the `lock` was in place specifically for `QueryMirrorDataAsyncTask` as per the comments, which has now been removed.

Reply:
I can check if the mutex is being used elsewhere and if it can be removed altogether.

Reply:
Ok, I just checked the `cachedValueMutex` is being used implicitly in the `CachedValue` getter and setter itself so it is safe to remove from here. No more to do here.